### ISSUE TITLE
CMP: rellocate fltk2 related references in code management plan to a separate FLTK2 doc

### ISSUE THREAD

**Issue by erco77** [fltk/fltk#187]
There are references to fltk2's coding techniques in the CMP that should probably be moved to a separate document, or at very least collected into a separate section instead of interwoven in the 1.x.

This would keep the CMP shorter, and prevent new devs from trying to do fltk2 style stuff in 1.x.

Example:
```
Function/Method/Variable Naming

All public (exported) functions and variables must be placed in the "fltk" namespace. [..]

Structure/Class Naming

All public (exported) structures and classes must be placed in the "fltk" namespace and consist of
capitalized words without the underscore, e.g. "fltk::SuperWidget". [..]
```

**Comment by Albrecht-S** [comment]
:+1: But please see private mail ...

**Comment by Albrecht-S** [comment]
What about moving the coding style docs from ["the CMP"](https://www.fltk.org/cmp.php) (Configuration Management Plan) into the FLTK docs and maintain the coding style there? This way the docs would be automatically "versioned" and future coding style changes could be done w/o affecting prior coding style (docs).

The question "where to put the FLTK 2 docs" still remains...

**Comment by erco77** [comment]
Regarding the fltk2 docs, a separate doc on the web only, kinda like how I relegated the old STR docs to a separate doc.

**Comment by MatthiasWM** [comment]
Just FYI, I have reserved www.fltk2.org to myself, I case we decide to split up the libraries...

**Comment by erco77** [comment]
On 1/21/23 09:48, Matthias Melcher wrote:

> Just FYI, I have reserved www.fltk2.org <http://www.fltk2.org> to myself, I case we decide to split up the libraries...
>

     So hope still springs eternal for fltk2, eh?

     I know there's folks who'd like to see it happen, and it has many design improvements,
     and has some really neat stuff like, I think, skinning, and a better designed menu system.

     It has a bad rep to get over, though. I know it miffed a few app developers who used it
     thinking designing for fltk2 would be forward thinking, but were left high and dry when
     it never released.

     Of course 1.4 is kinda late to release too, heh.. we've been pretty slow on getting that out.

**Comment by erco77** [comment]
Just please fix the site so it doesn't confuse people with the references to 1.x.

What we really don't want to have is a repeat of people thinking fltk2 is a thing, and start trying to ask about using it here on the 1.x project.

Until you have it working, I'd suggest having a page that doesn't even say what it is until you actually have something working and releasable. Otherwise I think it will only cause confusion in the public's eye.

**Comment by Albrecht-S** [comment]
> Just FYI, I have reserved [www.fltk2.org](http://www.fltk2.org/) to myself, I case we decide to split up the libraries...

OK, good step, but **PLEASE** replace the "official looking" but very badly formatted content with something that clearly states what this is for.

I also recommend to add a robots.txt to disable all search engines. With the link above it's only a question of time when they start indexing this new site 'fltk2 dot org' - and we should avoid that!

**Comment by Albrecht-S** [comment]
Regarding `robots.txt`:
```
User-agent: *
Disallow: /
```

... would disallow all search engines - as long as the site doesn't have any official content.

**Comment by Albrecht-S** [comment]
@erco77 Are our fltk.org newsgroups indexed? If yes, the link would sooner or later be picked by Google and others.

If it **is** indexed, would you please edit the contents of the news files as soon as possible ? I believe you could do this best (I could try but maybe it's better if you do it).

PS: I know that it's not good to change posted messages, but in this case I would suggest to do it.

**Comment by erco77** [comment]
By indexed you mean available to google?

AFAIK there's nothing blocking google from walking down into our web interface to the group, but it might not know how to navigate it.. not sure.

To check, I found a Jan 2021 posting on fltk.general, subject: "Re: Cropping SVG image fails", so I googled for:
```
    fltk "cropping svg image fails"
```
..and that did come up with hits to our group.

But um, I'm not clear though on what you're asking for here:

> If it is indexed, would you please edit the contents of the news files as soon as possible?
>  I believe you could do this best (I could try but maybe it's better if you do it).

..I'm not sure what you want me to edit.

**Comment by erco77** [comment]
Oh, reading back in this thread, you must mean Matt's references to his site from this github thread archived in the fltk.issues archive? Ya, I can edit those

**Comment by erco77** [comment]
OK, fixed in this thread and all the replies, e.g.
https://www.fltk.org/newsgroups.php?s5824+gfltk.issues+v5837

**Comment by Albrecht-S** [comment]
Yes, that was what I meant, sorry for being unclear and for being offline for a while. But I see you got it. And, also yes, my question was whether search engines would index the web interface to our newsgroups, and you answered this question as well. Looks good now, thanks.

**Comment by MatthiasWM** [comment]
Sorry, and seriously WTF. I have leased the domain, but I did not set up anything yet. I will check my Server to see why there is some content linked. Weird.

On the road right now, but will be home in an hour or so.

**Comment by MatthiasWM** [comment]
Oh, and no, I was not planning to revive fltk2. It was a just-in-case thing, and if anything, I would make a museum page out of it, leaving FLTK.org to 1.4 and maybe later 3.0, 4.0, or whatever we may call it (LOL, in case that was not clear).

**Comment by MatthiasWM** [comment]
Ping! Discussion drifted. 
1: AFIK, the CMP is not part of the Github repo, so accessing it is quite difficult. I suggest that:
1a: we slim it down considerably and make it part of this repo, or
1b: we generate a new repo under GitHub.com/fltk/ that contains these kinds of documents

2: I would love a vote on the fltk2.org URL. Do we want to move fltk2 into its own static and unmanaged web site and no longer reference it form fltk.org , or are we happy with the way it is

This issue will expire on 23 Nov 2025

**Comment by Albrecht-S** [comment]
This issue should be closed and we should discuss this topic (maintaining the CMP) off-list. There have been some efforts in the past, but this is really a long-standing issue...

**Comment by MatthiasWM** [comment]
Closed
